src/device/device_ledger.cpp

```diff
@@ -48,6 +48,15 @@ namespace hw {
     /* ===================================================================== */
     /* ===                           Debug                              ==== */
     /* ===================================================================== */
+    #ifdef WIN32
+    static char *pcsc_stringify_error(LONG rv) {
+     static char out[20];
+     sprintf_s(out, sizeof(out), "0x%08X", rv);
+
+     return out;
```

Comment:
This makes it non thread safe

Reply:
hmm. I just copy paste iDunk code :D
Looking for a full code

Reply:
You'd need something like static __thread char out[20] I guess, if that works.